### PR TITLE
Fix shutdown on Windows & process shutdown commands after main loop is  breaked

### PR DESCRIPTION
**Pull request by fabricecaruso72**

### CODE REVIEW

**Comment by tomaz82** [comment]
Great addition but I have 2 concerns.

First, why is it called exitMode when it's set by an SDL_QUIT, quitMode would be more logical?
And I fear those 3 defines are not safe given how events work in SDL. But I'd have to investigate that further.

A safer approach would be an userevent, with our own SDL_ES_QUIT defined to 0x8000, and the "mode" as code into that, 0 for restart, 1 for reboot and 2 for shutdown, just a suggestion, but the current approach is not safe, the SDL_MSG_REBOOT is currently as 0x2000 which already is SDL_RENDER_TARGETS_RESET.

**Comment by fabricecaruso** [comment]
I hesitated : The simpliest, safer & most evolutive solution would have been to store an independant from SDL int (or enum) exitMode ( or quitMode ) in platform.cpp, assigned in quitES, and refactor the code I added after main loop as a method processQuit in platform.cpp reading that exitmode and doing the job, SQL_QUIT being the pushed event in quitES... 
What do you think about it ?

Concerning the word quit/exit, it's better to normalize things, I was'nt careful bout that. Windows developpers all use the "exit" word by habit ( ie exitCode, ExitWindowEx... ), and I'm a Windows developper...

QUIT being the convention with SDL...

**Comment by tomaz82** [comment]
Having an entirely separate quitMode sounds like the safer option yes, all isolated from SDL.

**Comment by fabricecaruso** [comment]
> 
> 
> Having an entirely separate quitMode sounds like the safer option yes, all isolated from SDL.

Agree, I'll do that.

**Comment by fabricecaruso** [comment]
Ok it's done

File: es-app/src/main.cpp [hunk 1]
```diff
@@ -386,7 +386,7 @@ int main(int argc, char* argv[])
 			{
 				InputManager::getInstance()->parseEvent(event, &window);
 
-				if(event.type == SDL_QUIT)
+				if (event.type == SDL_QUIT)
```
Comment (tomaz82) [inline es-app/src/main.cpp]: This is an uncessesary change

**Comment by tomaz82** [comment]
Other than the not needed change in main.cpp this looks good to me.

**Comment by fabricecaruso** [comment]
> Other than the not needed change in main.cpp

The only change in main.cpp is the call to processQuitMode(); 
The others changed line consist in removing what I dit in my first commit

**Comment by jrassa** [comment]
> The only change in main.cpp is the call to processQuitMode();

There is second change that is just an unnecessary white space change.  Fix that and squash your commits into a single commit and this should be ready to merge.  Let me know if you need help with squashing.

**Comment by fabricecaruso** [comment]
> Let me know if you need help with squashing.

Mmm... Yeah... A little bit of help would be appreciated ;-)
I've never heard about squashing before ( too new on git )...

I'm currently trying to understand. I succeeded in merging my 4 commits into one, but in my master branch, not in fixed-shutdown branch. I still can't find a way to merge the 4 commits in the same branch, and I won't have time to do it today. I'll do it tomorrow. If you have tricks it would help ;-) 
Thanks.

**Comment by fabricecaruso** [comment]
Waaaah... I succedeed to do squash my commit... 
It's so not user-friendly !!!!!!
It remembered me the time when I was a student and I worked under unix with tools like Vi (in 1992)...

**Comment by tomaz82** [comment]
Looks good to me!

**Pull request merged**